src/v/pandaproxy/schema_registry/seq_writer.h

```diff
@@ -0,0 +1,88 @@
+//// Copyright 2021 Vectorized, Inc.
+////
+//// Use of this software is governed by the Business Source License
+//// included in the file licenses/BSL.md
+////
+//// As of the Change Date specified in that file, in accordance with
+//// the Business Source License, use of this software will be governed
+//// by the Apache License, Version 2.
+
+#pragma once
+
+#include "kafka/client/client.h"
+#include "pandaproxy/schema_registry/sharded_store.h"
+#include "pandaproxy/schema_registry/types.h"
+#include "random/simple_time_jitter.h"
+#include "utils/retry.h"
+
+namespace pandaproxy::schema_registry {
+
+using namespace std::chrono_literals;
+
+class seq_writer {
+public:
+    seq_writer(ss::sharded<kafka::client::client>& client, sharded_store& store)
+      : _client(client)
+      , _store(store) {}
+
+    ss::future<> read_sync();
+
+    ss::future<schema_id>
+    write_subject_version(subject sub, schema_definition def, schema_type type);
+
+private:
+    ss::future<> read_sync_inner();
+
+    ss::sharded<kafka::client::client>& _client;
+    sharded_store& _store;
+
+    struct WriteCollision : public std::exception {
```

Comment:
In general I've tried to put module exceptions in the `exceptions.h`, and inherited from `exception_base`, so it's easy to know what should be caught. But it's ok if this is all handled in this class, which it looks like it is. It might be worth adding some more context, but that's a nit.

Reply:
I've pulled this up into a `write_collision` error code alongside the other error codes used in `schema_proxy::exception`, so that it gets handled properly as an HTTP error code (503) if we reach the retry limit.